### DISCUSSION TITLE
Default object defined as dict, not object

### DISCUSSION THREAD

**Discussion by mtovt** [koxudaxi/datamodel-code-generator#797] [Q&A] [answered]
OAS says ([section 4.7.25.1 of the spec](https://spec.openapis.org/oas/v3.0.0#properties)):
> ... Unlike JSON Schema, the value MUST conform to the defined type for the Schema Object defined at the same level. For example, if type is string, then default can be "foo" but cannot be 1.

Example OAS:
```yml
openapi: 3.0.3
info:
  title: Example
  version: 0.1.0
servers:
- url: http://example.com
paths:
  /foo:
    delete:
      responses:
        '200':
          description: OK
components:
  schemas:
    Foo:
      type: object
      properties:
        text:
          type: string
          default: 987
        number:
          type: number
    Bar:
      type: object
      properties:
        foo:
          allOf:
          - $ref: '#/components/schemas/Foo'
          default:
            text: abc
            number: 123
```

Specification above generating following models
```python
class Foo(BaseModel):
    text: Optional[str] = 987
    number: Optional[float] = None


class Bar(BaseModel):
    foo: Optional[Foo] = {'text': 'abc', 'number': 123}
```

I expected something like:
```python
class Foo(BaseModel):
    text: Optional[str] = 987
    number: Optional[float] = None


class Bar(BaseModel):
    foo: Optional[Foo] = Foo(text='abc', number=123)
```

Why are default values used in a dict for now? Seems default values from this example should be generated as an instance of class `Foo`.

**Comment by koxudaxi** [comment, marked as answer]
@mtovt I'm sorry for my too-late reply.
OK, I will support the feature.
Thank you!!

**Reply by koxudaxi** [reply]
I just created the issue.
https://github.com/koxudaxi/datamodel-code-generator/issues/1024

**Reply by koxudaxi** [reply]
I have released a new version `0.17.0` 
The version supports object and list default.

Examples
```python
class Foo(BaseModel):
    text: Optional[str] = 987
    number: Optional[float] = None


class Bar(BaseModel):
    foo: Optional[Foo] = Field(
        default_factory=lambda: Foo.parse_obj({'text': 'abc', 'number': 123})
    )
    baz: Optional[List[Foo]] = Field(
        default_factory=lambda: [
            Foo.parse_obj(v)
            for v in [{'text': 'abc', 'number': 123}, {'text': 'efg', 'number': 456}]
        ]
    )
```
https://github.com/koxudaxi/datamodel-code-generator/blob/master/tests/data/expected/main/main_openapi_default_object/__init__.py
